crates/iceberg-catalog/src/implementations/kv2.rs

```diff
@@ -0,0 +1,307 @@
+            ))
+        })?;
+
+        Ok(Secret {
+            secret_id: *secret_id,
+            secret: vaultrs::kv2::read_version::<S>(
+                &*self.vault_client.read().await,
+                self.secret_mount.as_str(),
+                &secret_ident_to_key(*secret_id),
+                metadata.current_version,
+            )
+            .await
+            .map_err(|err| {
+                IcebergErrorResponse::from(ErrorModel::internal(
+                    "secret read failure",
+                    "SecretReadFailed",
+                    Some(Box::new(err)),
+                ))
+            })?,
+            created_at: metadata.created_time.parse().map_err(|err| {
+                IcebergErrorResponse::from(ErrorModel::internal(
+                    "secret metadata read failure",
+                    "SecretReadFailed",
+                    Some(Box::new(err)),
+                ))
+            })?,
+            updated_at: Some(metadata.updated_time.parse().map_err(|err| {
+                IcebergErrorResponse::from(ErrorModel::internal(
+                    "secret metadata read failure",
+                    "SecretReadFailed",
+                    Some(Box::new(err)),
+                ))
+            })?),
+        })
+    }
+
+    /// Create a new secret
+    async fn create_secret<S: Send + Sync + Serialize + std::fmt::Debug>(
+        &self,
+        secret: S,
+    ) -> Result<SecretIdent> {
+        let secret_id = SecretIdent::from(Uuid::now_v7());
+        vaultrs::kv2::set(
+            &*self.vault_client.read().await,
+            self.secret_mount.as_str(),
+            &secret_ident_to_key(secret_id),
+            &secret,
+        )
+        .await
+        .map_err(|err| {
+            ErrorModel::internal(
+                "secret creation failure",
+                "SecretCreationFailed",
+                Some(Box::new(err)),
+            )
+        })?;
+        Ok(secret_id)
+    }
+
+    /// Delete a secret
+    async fn delete_secret(&self, secret_id: &SecretIdent) -> Result<()> {
+        Ok(vaultrs::kv2::delete_metadata(
+            &*self.vault_client.read().await,
+            self.secret_mount.as_str(),
+            &secret_ident_to_key(*secret_id),
+        )
+        .await
+        .map_err(|err| {
+            ErrorModel::internal(
+                "secret deletion failure",
+                "SecretDeletionFailed",
+                Some(Box::new(err)),
+            )
+        })?)
+    }
+}
+
+fn secret_ident_to_key(secret_id: SecretIdent) -> String {
+    format!("secret/{secret_id}", secret_id = secret_id.as_uuid())
+}
+
+impl std::fmt::Debug for SecretsState {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("SecretsState")
+            .field("vault_client", &"VaultClient")
+            .field("secret_mount", &self.secret_mount)
+            .field("vault_user", &self.vault_user)
+            .field("vault_password", &"REDACTED")
```

Comment:
We are using veil in other places to redact - should we use it here as well?

Reply:
vault_client doesn't impl Debug, cursory read of veils docs makes me think we cannot use it here since it derives Debug for the struct?